metadata/metadata.go

```diff
@@ -337,34 +337,34 @@ func (lookup localResourceLookup) SetSchedule(id ResourceID, schedule string) er
 	return nil
 }
 
-func (lookup localResourceLookup) HasJob(id ResourceID) (bool, error) {
+func (lookup LocalResourceLookup) HasJob(id ResourceID) (bool, error) {
 	return false, nil
 }
 
-type sourceResource struct {
+type SourceResource struct {
 	serialized *pb.Source
 }
 
-func (resource *sourceResource) ID() ResourceID {
+func (resource *SourceResource) ID() ResourceID {
 	return ResourceID{
 		Name: resource.serialized.Name,
 		Type: SOURCE,
 	}
 }
 
-func (resource *sourceResource) Schedule() string {
+func (resource *SourceResource) Schedule() string {
 	return ""
 }
 
-func (resource *sourceResource) Dependencies(lookup ResourceLookup) (ResourceLookup, error) {
-	return make(localResourceLookup), nil
+func (resource *SourceResource) Dependencies(lookup ResourceLookup) (ResourceLookup, error) {
+	return make(LocalResourceLookup), nil
 }
 
-func (resource *sourceResource) Proto() proto.Message {
+func (resource *SourceResource) Proto() proto.Message {
 	return resource.serialized
 }
 
-func (this *sourceResource) Notify(lookup ResourceLookup, op operation, that Resource) error {
+func (this *SourceResource) Notify(lookup ResourceLookup, op operation, that Resource) error {
```

Comment:
super nit: i know you didn't write this but since you're refactoring, golang best practices is to use the same receiver name everywhere. This one's kinda special because it uses "this" and "that" to differentiate the two. I'd still probably keep it consistent for readability but super minor just wanted to point it out

Reply:
I noticed that warning too, but wasn't sure if changing it would wreck something. Looks like it's safe to change. Swapping now since it's quick...I think. 
![image](https://github.com/featureform/featureform/assets/34227093/fb9eece2-0b3a-4cbd-a687-b77550622d22)

Reply:
ok renamed the `that` param to `resourceParam` and renamed the `this` receiver to `resource`.

Reply:
I triple checked but give it a scan when you can.